### DISCUSSION TITLE
Ender-3 V2/S1 Professional Firmware 20230805

### DISCUSSION THREAD

**Discussion by mriscoc** [mriscoc/Ender3V2S1#1010] [Announcements]
AUGUST 2023
=========

## What's new in this release ##
- LCD brightness dims after 5 minute timeout (not available in all versions)
- Slow retract item in Filament management menu
- Better bed tramming wizard
- Automatic homing axes before M48 (Probe test) if necessary
- Support to M86-M87 Hotend Idle Timeout
- Fixed C104 bug
- Fixed bug in UBL when an invalid point is found
- Improved UBL smart-fill
- M300 E1/0 enables or disables sounds
- C851 F edit probe Z fix compensation
- Fixed BLTouch deploy bug in HS mode
- Several minor bug fixes and Misc. optimizations
- New OFFICIAL/CUSTOM BUILD label in the info page
- and many more fixes from latest Marlin's bugfix 2.1.x

In Source:
- Added support to One cLick print https://github.com/MarlinFirmware/Marlin/pull/25781
- Preliminary support for E3 Free-runs board CR4NTxxC10 and other boards https://store.creality.com/products/e3-free-runs-tmc2209-32-bit-open-source-silent-motherboard
- Fixed Time Motion experimental support
- Improved support for Marlin Auto Build firmware xfer
- Maple support was deprecated

> **Note**1: In latest releases I'm promoting UBL versions over ABL, but ABL versions will be available through custom compile. If you had a `G29` to make an ABL leveling before each print change it to `G29 P1` for UBL. MPC is also being incorporated in replacement of the hot-end PID which will continue to be available on custom compiles.

> **Note**2: Due to lack of program memory space in the STM32**F4**01RCT6 SoC some features had to be disabled for the F4 UBL firmware version, for example: BAUD_RATE_GCODE, INDIVIDUAL_AXIS_HOMING_SUBMENU, SOUND_MENU_ITEM, LONG_FILENAME_WRITE_SUPPORT, BINARY_FILE_TRANSFER, POWER_LOSS_RECOVERY.

## CV LASER MODULE SUPPORT for Ender-3S1!!
https://github.com/mriscoc/Special_Configurations/releases/tag/CVLM
![laser-small](https://github.com/mriscoc/Ender3V2S1/assets/2745567/a9b58e8f-d646-443c-93a2-c1200dc0844a)

This firmware is based on [Marlin](https://github.com/MarlinFirmware/Marlin), has been optimized and has enhancements for the Creality Ender 3 V2/S1 printer. This is an **universal firmware** that can be used in both RET6 and RCT6 STM32 chips that have 256k or 512k Flash memory. The firmware and its wiki are not really intended for beginners but as a supplement to already existing guides.  

[How to install the firmware](https://github.com/mriscoc/Ender3V2S1/wiki/How-to-install-the-firmware)  
   
<img align="left" src="https://raw.githubusercontent.com/mriscoc/Ender3V2S1/Ender3V2S1-Released/screenshots/main.jpg" width="270" height="480" />
<img  src="https://raw.githubusercontent.com/mriscoc/Ender3V2S1/Ender3V2S1-Released/screenshots/main-iconset.jpg" width="270" height="480" />

## Requirements
In order to use this firmware, your printer must meet the following requirements:
- A compatible control board: Creality 4.2.2, 4.2.3, 4.2.7, V24S1_301 or SKR Mini E3 V3 for Ender series printers with a STM32 SoC.
- A DWIN, DACAI, TJC or SYNWIT color display with encoder knob. Touch or legacy monochrome screens are not supported.

*You can modify your printer to meet these requirements*. 

## Versions
- **Ender 3v2 MM**: if you have the Ender 3v2 near to stock then use this version (Manual Mesh).
- **Ender 3v2 BLT**: use this version if you installed a CR/3D/BLTouch level sensor in the Ender 3v2.
- **Ender 3S1**: use this version if you have an Ender 3S1 printer with CRTouch, (F1 is for the STM32F103 and F4 is for the STM32F401 SoC)

You can download the __latest binaries__ from the assets at the bottom of
the __[release page](https://github.com/mriscoc/Ender3v2/releases/latest)__.
To know if you have the last release go to the info page in the display of
your printer and compare the release build date with the name of the binary
file. The filename follows the convention:  
  
> Ender3V2-422-MM-**YYYYMMDD**.bin
  
Where YYYYMMDD indicates the date of the build. 422 refers to the mainboard
version **4.2.2**, 423 refers to the mainboard version **4.2.3**, and 427 is for the **4.2.7** mainboard version, if you got
a **4.3.1** board use the 422 version. SKRMINI version is currently for the
BTT SKR Mini E3 V3 board (needs a screen adapter).

For example, if you want a firmware for your Ender-3 V2 printer with a control
board version 4.2.2, CRTouch and support for UBL, you should download the file:

> Ender3V2-422-BLTUBL-YYYYMMDD.bin

![Identificando_version_de_placa](https://user-images.githubusercontent.com/2745567/168324748-0c6dca47-41f3-4ad5-9146-a1a3caf6a7ce.jpg)

### UBL versions
Binaries with UBL in their filename have support for [Marlin Unified Bed Leveling](https://marlinfw.org/docs/features/unified_bed_leveling.html), this system is similar to the bilinear leveling but with a few extra features to provide users with more control. There are multiple guides on the web about UBL, we recommend reading about it before using this new leveling system.

* https://marlinfw.org/docs/features/unified_bed_leveling.html
* https://all3dp.com/2/marlin-ubl-unified-bed-levelin-simply-explained/
* https://www.3dmakerengineering.com/blogs/3d-printing/unified-bed-leveling-marlin
* https://hackaday.com/2022/01/14/3d-printering-getting-started-with-universal-bed-leveling/

### Model Predictive Temperature Control ([MPC](https://marlinfw.org/docs/features/model_predictive_control.html))
From this version [MPC](https://github.com/mriscoc/Special_Configurations/releases/tag/MPC) is being incorporated in all versions. MPC has proven to be a better algorithm for keeping the nozzle temperature stable, and is also very useful for high power heaters.

## Special versions
There are several special configurations here: https://github.com/mriscoc/Special_Configurations/releases

- [Linear Advance (LA) versions](https://github.com/mriscoc/Special_Configurations/releases/tag/LinearAdvance)
- [Input Shaping (IS) versions](https://github.com/mriscoc/Special_Configurations/releases/tag/InputShaping)
- [Marlin #5 temperature sensor V6/Volcano](https://github.com/mriscoc/Special_Configurations/releases/tag/T5)
- [Marlin #13 (3950) temperature sensor 300°C support (Sprite Extruder Pro)](https://github.com/mriscoc/Special_Configurations/releases/tag/T13)

## Community links
* [Telegram](https://t.me/ender3v2s1firmware)
* [Reddit](https://www.reddit.com/r/Ender3v2Firmware) 
* [Ender 3V2 Facebook](https://www.facebook.com/groups/ender3v2firmware)
* [Ender 3S1 Facebook](https://www.facebook.com/groups/ender3s1printer)

## Donations
Thank you for your support, I receive donations through [Patreon](https://www.patreon.com/mriscoc) and [Paypal](https://www.paypal.com/paypalme/mriscoc)   

[<img src="https://www.paypalobjects.com/en_US/i/btn/btn_donateCC_LG.gif">](https://www.paypal.com/donate?business=85SPAAR6UZEE8&currency_code=USD)

# Firmware features

* Universal RCT6/RET6 Ender 3 V2/S1 Support
* DWIN / DACAI / TJC / SYNWIT display support
* Improved support for Manual Mesh and CR/3D/BLTouch
* User configurable power lost recovery enabler
* Manual and assisted bed tramming
* Tramming Wizard for BL/3D/CRTouch versions
* Long filename support with scroll
* Axis value in dash board shows actual position of the axes, including Z leveling
* Baby Z-Stepping
* Firmware Retract
* Advanced pause feature
* Configurable park-head position (default x=240 y=220)
* Filament change support
* Filament management (load/unload/park head) in prepare menu
* Print Flow on Motion and Tune menu 
* Home axis after motor disabling
* Fastest Z feed rate
* Live move for all motors but extruder
* Support for S-curve Acceleration
* Reboot printer item on control menu
* Increase extruder feed rate limit to 150 for BMG users
* Allows to invert direction of the extruder on Motion menu
* Support for BTT Smart filament sensor
* Selectable runout filament sensor active state (LOW/HIGH/MOTION)
* Editable runout distance
* Support for memory card extensions via Media Update (auto-mount) Enable/Disable menu item
* PID popup with temperature plot
* Lock screen
* Editable physical dimensions
* Mesh viewer, leveling progress and editor
* Custom pre-heat
* Enable/disable beeps sound
* View of g-code embedded thumbnail (use [this ](https://github.com/mriscoc/Ender3V2S1/wiki/How-to-generate-a-gcode-preview)guide)
* New print done screen
* Two decimal positions for axis coordinates value
* New Z-Offset wizard
* New PID settings menus
* PID moves the head to the center for more accuracy
* Leveling bed temperature linked to bed PID temperature
* Editable extrude minimum temperature
* Brightness settings on tune menu
* Turn off the backlight of the lcd in the tune menu
* New status line system, allows the scroll of large messages
* Home submenu with individual axis homing
* Enabled user setting for 115200 bauds (Creality Box compatibility)
* Head parking before recovery after power failure
* Offset icon changes to show leveling status
* Live end-stop diagnostic page
* Move bed to front after cancelling a print.
* Editable firmware retracts settings for Tune and filament settings menu
* Print Statistics page accessible from the Advanced Settings menu
* Toolbar with 11 configurable shortcuts actions.
* Support in source code of control menus for LED lights as the Neopixel
* Support up to 10 Preheat settings (needs own compile)
* Support for Model predictive temperature control "MPC" (needs own compile)
* Support for Linear Advance and Input Shaping in the [Special Configuration repository](https://github.com/mriscoc/Special_Configurations)
* TMC Trinamic driver current editor menu when it is supported (not for standalone)
* Menu item for enable/disable Adaptive Step Smoothing when it is supported

![latest release](https://raw.githubusercontent.com/mriscoc/Ender3V2S1/Ender3V2S1-Released/screenshots/New-beta.gif)

### Manual mesh
* Assisted manual bed tramming
* Assisted manual mesh leveling menu
* Selectable mesh points from 2x2 to 9x9
* Mesh viewer
* Persistent Z-offset
* Move Z to 0 and disable motors option
* Mesh editor for fine tuning
* Allows to set the Z height after Z home

### CR/3D/BLTouch
* Safe Probe Z-Homing
* Bed auto level
* High speed probe
* Retry bed level on error
* Mesh viewer
* Assisted tramming
* Selectable mesh points from 2x2 to 9x9
* Allows the user to cancel an auto leveling process 
* Increase Max Probe Offset and Probing Margin to 60 mm
* M48 Probe Test to Probe Settings menu
* Probe Stow/Deploy in Probe Settings menu
* Option to disable HS mode for better compatibility with CRTouch
* Option to change the Z Probe feed rate
* Configurable Bed auto leveling temperature
* Mesh editor for fine tuning
* Automatic maximize leveling area
* Automatic center leveling area
* Editable multiple probing
* Leveling mesh creation progress view

<a href="https://youtu.be/qzYJC2AHbW8"><img src="https://img.youtube.com/vi/qzYJC2AHbW8/maxresdefault.jpg" width="640" height="400" /></a>  
  
### G-Codes enhanced support
* Filament Load Unload G-codes (M701/M702)
* Support for M600 filament change
* Support for G-code M117 messages
* Enable M575 change baud rate by G-code
* Extended M75 with filename support, use M75 {{ event.name }} in g-code script
* Enable ARC and 'P' parameter to specify complete circles (G2/G3)
* G35 support
* M155 Geometry report
* Native support of marlin brightness and M256 g-code

### Support for special nonstandard custom C G-Codes
* C10 marks a G-code file as a [configuration file](https://github.com/mriscoc/Ender3V2S1/wiki/Configuration-files) to avoid confirm to print and print done dialogs.
* Implemented C11 En Rn Gn Bn gcode for colorize UI elements
* C108: Cancels screens waiting for user simple confirmation
* C29 F### B### L### R### N# T#:  Mesh insets, amount of grid points and bed temperature of the mesh leveling
* C100 X### Y###: to sets minimum limits
* C101 X### Y### z###: sets maximum limits
* C102 X### Y###: sets bed size
* C104 U1 T###: unlocks the hotend maximum temperature to ###°C (CAUTION) 
* C115: Shows ProUI details
* C125 X### Y### z###: sets the park position
* C412 M#: sets run-out sensor active state (M0:LOW, M1:HIGH, M2:MOTION)
* C510: locks/unlocks the screen (C510 U1 to unlock)
* C562 E#: inverts the Extruder (E1 to invert)
* C810: sets up the toolbar shortcuts
* C851 S#### [Z###] Mn: sets probe Z feed rate (S) and set Multiple probing (Mn>1) or disables it (M0), in manual mesh
version, Z parameter can be used to set a manual Z-offset.

<img src="https://raw.githubusercontent.com/mriscoc/Ender3V2S1/Ender3V2S1-Released/themes/Select_Colors.jpg" width="270" height="480" /> 

### Advanced settings menu
* Advanced mesh leveling editing features
* Probe offsets menu (BLTouch version)
* Customizable park-head position
* Customizable bed size and z heigh up to 999 x 999 x 999
* Launch PID for nozzle and bed with graphical feedback
* Filament run-out sensor with active state setup (LOW/HIGH)
* Filament load and unload lengths menu items in run-out menu
* Power-loss recovery enable/disable option
* LCD Brightness control
* Select Color Sub Menu
* Beep/Sound enable/disable option
* Selectable mesh point (BLTouch version)
* Media Update (auto-mount) for memory card extensions support
* Lock screen
* End-stop diagnostics

### Special Octoprint / Host print support
* High speed (250000 baud) serial communication
* In screen control of host printing (M73, M75-M77) 
* Display messages on status line (M117)
* Color change of UI elements for visual status report (C11)
* Binary file transfer for remote firmware upgrade
* List binary (.bin) files (M20) for flash firmware support with BFT
* Arc Welder Octoprint plugin support
* EEPROM Editor Octoprint plugin support
* Host action commands for filament runout
* Emergency Parser active
* Advanced OK

**Octoprint bed visualizer support**
<img src="https://raw.githubusercontent.com/mriscoc/Ender3V2S1/Ender3V2S1-Released/screenshots/Bed-visualizer.png" width="350" height="225" />

### Display enhancements
* Extended DWIN library
* Enhanced status messages
* Improved status bar with temperature, flow, fan, z-offset and xyz positions
* MPC/PID auto tune progress on display with plot
* Leveling progress on display
* Boot screen for M997
* Transparent icons for better color theme adaptability
* New print g-code preview
* New print done screen

**A lot of additions and fixes from Bugfix development branch**

## More info on the wiki pages ##
- [**How to install the firmware**](https://github.com/mriscoc/Ender3V2S1/wiki/How-to-install-the-firmware)

- [**Octoprint settings**](https://github.com/mriscoc/Ender3V2S1/wiki/Octoprint)

- [**How to make the Manual Mesh**](https://github.com/mriscoc/Ender3V2S1/wiki/Manual-Mesh)

- [**How to install a 3D/BLTouch**](https://github.com/mriscoc/Ender3V2S1/wiki/3D-BLTouch)

- [**How to apply a color theme**](https://github.com/mriscoc/Ender3V2S1/wiki/Color-Themes)

https://user-images.githubusercontent.com/2745567/143180943-f69f5967-34c1-421f-86a1-466ecccab97d.mp4

<hr /><em>This discussion was created from the release <a href='https://github.com/mriscoc/Ender3V2S1/releases/tag/20230805'>Ender-3 V2/S1 Professional Firmware 20230805</a>.</em>

**Comment by Siggy101** [comment]
Thank you for a fresh new release.
It's typical that I decided that yesterday was the day I would take the time to compile myself a firmware bin for my printer!!
I guess I get to do it all again today :)

**Comment by bidibulle38** [comment]
Thx for the fresh release!
I was actually trying out that firmware today and had just updated to the May release.
A couple bugs I found compared to the Jyers is that the "bed heating" text does not display (just "printing") and neither does the degree symbol next to temperatures. The M140_M190.cpp has the expected reference to MSG_BED_HEATING as it should and for the degree I looked and found a suggestion to disable USE_SMALL_INFOFONT, but that seems already done. Weird :-(
(These could be issues with Marlin source, but I dont know where to do a bug report either way) 
/Ender3v2+4.2.2MB

**Comment by steve28** [comment]
Sorry - new around here.... will there be a TJC display one?  I have the S1 with a TJC display.  Thanks!

**Reply by mriscoc** [reply]
For TJC support: https://github.com/mriscoc/Special_Configurations/releases/tag/tjc

**Reply by steve28** [reply]
Thank you!  I had checked yesterday right before I posted, but I guess I should have been more patient! 😬

**Reply by mriscoc** [reply]
I'm able to find bugs thanks to people like you doing early testing, it's impossible for me to test all binaries on all kinds of boards.

**Comment by classicrocker883** [comment]
@mriscoc I was able to Merge with the most recent *Marlin* so that now you can view all ~19,000+ commits. this is helpful say when you make the next release, so that everyone is able to see the changes and can refer to Marlin commits if need to.

[link to repo with merge]()

all you need to do is download the source code .zip file or *git clone* the repository, and enter `git reset --hard 6ce3545` or `ffef8f1` to revert back to this moment and continue where you left off, or the current most recent Marlin commit. after that just **PULL from Branch** with *MarlinFirmware/Marlin* **bugfix-2.1.x** to update with newer Marlin fixes.

**Reply by mriscoc** [reply]
Please do not post links to unofficial repositories.
I have a development repository with all the commits. I prefer to keep the public repository as simple as possible.

**Comment by 4X1RV** [comment]
Hello
Update goes into loop (new version label showing repeating every 10 seconds or so)
Taken the SD out it boots into the new version but with a SD (without the .BIN) it is stuck at booting.

Went back to former version,   works OK.

**Reply by mriscoc** [reply]
Format or change your SD card and try again.

**Comment by Dakweed666** [comment]
Just updated my 3 V2 Neo, all good I used the updated compile for the TJC display, it doesn't have the filament change, load or unload feature in the menu, is this normal?

**Reply by mriscoc** [reply]
For load/unload filament: 
Prepare / Filament management

**Comment by Dakweed666** [comment]
Thank you very much for the reply, it is all working now. Appreciate your work. Cheers, Jeff. Sent from my Galaxy
-------- Original message --------From: Miguel Risco-Castillo ***@***.***> Date: 14/08/23  6:22 am  (GMT+12:00) To: mriscoc/Ender3V2S1 ***@***.***> Cc: Dakweed666 ***@***.***>, Comment ***@***.***> Subject: Re: [mriscoc/Ender3V2S1] Ender-3 V2/S1 Professional Firmware 20230805 (Discussion #1010) 
Format you change your SD card and try again.

**Comment by 4X1RV** [comment]
Hi
Good advice from this forum,   check the setting for the 'filament runout' and set it to 'NO'  
The setting was 'ON' and caused rebooting all the time. 
Ender3V2 422 MB.

Then check other settings and everything works fine!

**Reply by MagicStuffCL** [reply]
I had this same problem, thanks for the solution.

**Reply by mriscoc** [reply]
Check https://github.com/mriscoc/Ender3V2S1/issues/1025

**Comment by CuparGuy** [comment]
Ok I'm completely new to all this.  But can this be used on my Ender 3 max neo?  I've looked at making my own but it's so confusing.

**Comment by notthepotato0** [comment]
I need a little help, for whatever reason my printer keeps pausing every 10 seconds when I try to print something and it tells me that there is an advanced paused because of the filament sensor, I tried it both plugged and unplugged, please help....

**Reply by mriscoc** [reply]
Check https://github.com/mriscoc/Ender3V2S1/issues/1025

**Comment by 4X1RV** [comment]
HiDid you try disabling the sensor   (NO).(in advanced,filament setting, enable runout to 'no' or empty square).

Or try changing the 'runout' setting (motion or HIGH or LOW)

    On Saturday, August 19, 2023 at 08:38:14 AM GMT+3, notthepotato ***@***.***> wrote:  
 
 


I need a little help, for whatever reason my printer keeps pausing every 10 seconds when I try to print something and it tells me that there is an advanced paused because of the filament sensor, I tried it both plugged and unplugged, please help....

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you commented.Message ID: ***@***.***>
  

|  | Virus-free.www.avast.com |

**Comment by Magutanko** [comment]
Hey, 
I've been trying to compile a firmware for my ender extender kit using the "Ender3V2-422-BLT" configurations. 
I've updated only my bed size  (X400,Y400,Z450), yet when using VSCode and PlatformIO to compile, I get consistent errors along the lines of:

error: missing binary operator before token "("
  273 | #if EITHER(PARKING_EXTRUDER, MAGNETIC_PARKING_EXTRUDER)

(about 8 similiar in total)
I'm absolutely new to all this, and really have no idea what I'm doing. I just want to get my printer working, and im sort of learning as i go. 

Thanks for any advice etc, 
This firmware seems amazing, I'd love to get it working!

Charlie

**Reply by mriscoc** [reply]
You don't need to do a custom build just to increase print volume. Use a precompiled one and this guide: https://github.com/mriscoc/Ender3V2S1/wiki/Calibration-Guides

**Reply by Magutanko** [reply]
Ah! I'm a fool, of course!
Apologies, I've spent all night trying to get my printer fixed, I think my brain is fried!

Cheers,

**Comment by vreyes508** [comment]
Hi Everyone, 
 
I am attempting to update my firmware for my ender 3S1, with the CR-touch, this would be the correct firmware for my S1? 
>[Ender3S1-F4-UBL-20230810.bin](https://github.com/mriscoc/Ender3V2S1/releases/download/20230805/Ender3S1-F4-UBL-20230810.bin)

**Reply by mriscoc** [reply]
If your previous firmware was stock, maybe the [Ender3S1-F4-20230809.bin](https://github.com/mriscoc/Ender3V2S1/releases/download/20230805/Ender3S1-F4-20230809.bin) will make you feel more comfortable.

**Reply by vreyes508** [reply]
Hi
mriscoc,
my previous version was 3.03. firmware from creality.
does the [Ender3S1-F4-20230809.bin](https://github.com/mriscoc/Ender3V2S1/releases/download/20230805/Ender3S1-F4-20230809.bin) firmware already have the CR-touch enabled?

**Reply by mriscoc** [reply]
Yes, all F1 and F4 firmware have CRTouch support.

**Reply by vreyes508** [reply]
The Daemon lite tool, 
![image](https://github.com/mriscoc/Ender3V2S1/assets/69439641/9727791b-8430-4077-95a9-05b2ed5aaee1)
I have formatted my card (Not the original card for my S1), 3 times the firmware still hasn't changed, do I need to unmount the image or?  
I do have Daemon tool lites agent, I when open the bin file i am getting this 
![image](https://github.com/mriscoc/Ender3V2S1/assets/69439641/1c5a2bfd-597f-496f-9b63-0f5b00e34b93)
 
What I am doing wrong here?

**Reply by mriscoc** [reply]
Use a SD card with a partition of 16GB or lower, formatted in MBR, FAT32 and 4096 allocation unit. Check the wiki for instructions about F4 installation procedure.

**Reply by vreyes508** [reply]
Hi Mriscoc,

I have done this proceedure the firmware is still not installing on my S1

**Comment by clrozeboom** [comment]
HI, @mriscoc, does the special configuration of the Ender3S1-F4-TJC support UBL or does that require a special build of the code? I am using this TJC build and it has mesh creation menus which build a usable mesh, but it is gone after any reboot of the machine. Is this a limitation of the S1 that there is no room to save a mesh with this special TJC support build? When I was using the standard Ender3S1-F4-UBL build, I thought it did save the mesh for reload later. Thanks!

**Reply by mriscoc** [reply]
Only binaries with UBL in the file name have UBL support, others have ABL, in both the leveling data is saved, but in UBL the data persists across firmware updates.

After create the mesh, you need to save settings of press the Save button in the mesh viewer at the end of mesh creation.

**Comment by MRichter187** [comment]
I understand this is an older thread but we're new to this firmware and I can't figure out how to enable power loss recovery

**Reply by mriscoc** [reply]
Power Outage item in Advanced Menu, not available in Ender 3S1 F4 printers
